fedimint-load-test-tool/src/main.rs

```diff
@@ -0,0 +1,627 @@
+        )]
+        generate_invoice_with: Option<LnType>,
+    },
+}
+
+#[derive(Debug, Clone)]
+struct Event(String, Duration);
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    fedimint_logging::TracingSetup::default().init()?;
+    let opts = Opts::parse();
+    let (event_sender, mut event_receiver) = tokio::sync::mpsc::unbounded_channel();
+    let futures = match opts.command {
+        Command::TestDownload { connect } => {
+            test_download_config(&connect, opts.users, event_sender.clone()).await?
+        }
+        Command::LoadTest {
+            connect,
+            initial_notes,
+            generate_invoice_with,
+            gateway_public_key,
+        } => {
+            let connect = if let Some(connect) = connect {
+                connect.to_owned()
+            } else {
+                cmd!("fedimint-cli", "connect-info").out_json().await?["connect_info"]
+                    .as_str()
+                    .map(ToOwned::to_owned)
+                    .expect("connect-info command to succeed")
+            };
+            let initial_notes = if let Some(initial_notes) = initial_notes {
+                initial_notes.to_owned()
+            } else {
+                cmd!("fedimint-cli", "spend", "2000sats").out_json().await?["note"]
+                    .as_str()
+                    .map(parse_ecash)
+                    .transpose()?
+                    .expect("spend command to succeed")
+            };
+
+            let gateway_public_key = if let Some(gateway_public_key) = gateway_public_key {
+                Some(gateway_public_key)
+            } else if let Some(generate_invoice_with) = generate_invoice_with {
+                Some(get_gateway_public_key(generate_invoice_with).await?)
+            } else {
+                None
+            };
+            if generate_invoice_with.is_none() {
+                warn!("No --generate-invoice-with given, not LN/gateway tests will be run")
+            }
+            run_economy(
+                opts.users,
+                connect,
+                initial_notes,
+                generate_invoice_with,
+                gateway_public_key,
+                event_sender.clone(),
+            )
+            .await?
+        }
+    };
+    let summary_handle = tokio::spawn(async move {
+        let mut results = HashMap::new();
+        while let Some(event) = event_receiver.recv().await {
+            let entry = results.entry(event.0).or_insert_with(Vec::new);
+            entry.push(event.1);
+        }
+        for (k, mut v) in results {
+            v.sort();
+            let len = v.len();
+            let max = v.iter().last().unwrap();
+            let min = v.first().unwrap();
+            let median = v[len / 2];
+            let sum: Duration = v.iter().sum();
```

Comment:
Nit: since you already need to `iter()` whole thing, you could `.fold` to `(sum, min, max)` in one go.